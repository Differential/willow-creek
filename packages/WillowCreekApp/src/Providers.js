import { nest } from 'recompose';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

export default nest(ClientProvider, Providers);
