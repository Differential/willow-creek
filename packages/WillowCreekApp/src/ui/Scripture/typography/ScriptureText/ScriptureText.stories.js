import React from 'react';
import { View } from 'react-native';
import { storiesOf } from '@storybook/react-native';

import ScriptureText from '.';

storiesOf('Scripture/typography/ScriptureText', module)
  .add('Regular', () => (
    <ScriptureText>
      {`"God's commands are designed to guide you to life's very best. You will not obey Him, if you do not believe Him and trust Him. You cannot believe Him if you do not love Him. You cannot love Him unless you know Him." ― Henry Blackaby`}
    </ScriptureText>
  ))
  .add('Bold', () => (
    <ScriptureText bold>
      {`"God's commands are designed to guide you to life's very best. You will not obey Him, if you do not believe Him and trust Him. You cannot believe Him if you do not love Him. You cannot love Him unless you know Him." ― Henry Blackaby`}
    </ScriptureText>
  ))
  .add('Italic', () => (
    <ScriptureText italic>
      {`"God's commands are designed to guide you to life's very best. You will not obey Him, if you do not believe Him and trust Him. You cannot believe Him if you do not love Him. You cannot love Him unless you know Him." ― Henry Blackaby`}
    </ScriptureText>
  ))
  .add('Bold Italic', () => (
    <ScriptureText bold italic>
      {`"God's commands are designed to guide you to life's very best. You will not obey Him, if you do not believe Him and trust Him. You cannot believe Him if you do not love Him. You cannot love Him unless you know Him." ― Henry Blackaby`}
    </ScriptureText>
  ))
  .add('isLoading', () => (
    <ScriptureText isLoading>
      {`"God's commands are designed to guide you to life's very best. You will not obey Him, if you do not believe Him and trust Him. You cannot believe Him if you do not love Him. You cannot love Him unless you know Him." ― Henry Blackaby`}
    </ScriptureText>
  ))
  .add('Border Box – platform testing', () => {
    const border = { borderWidth: 1, borderStyle: 'solid' };

    return (
      <View>
        <ScriptureText style={border}>Body Text</ScriptureText>
        <ScriptureText style={border}>
          {`"God's commands are designed to guide you to life's very best. You will not obey Him, if you do not believe Him and trust Him. You cannot believe Him if you do not love Him. You cannot love Him unless you know Him." ― Henry Blackaby`}
        </ScriptureText>
      </View>
    );
  });
