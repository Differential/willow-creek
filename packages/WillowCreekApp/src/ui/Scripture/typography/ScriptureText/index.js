import { Text } from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure, setDisplayName } from 'recompose';

import { styled, withPlaceholder, Placeholder } from '@apollosproject/ui-kit';

const { Typography } = Placeholder;

const styles = styled(({ theme, bold, italic }) => {
  let fontStack = theme.typography.serif.regular.default;

  if (bold && italic) {
    fontStack = theme.typography.serif.bold.italic;
  } else if (bold) {
    fontStack = theme.typography.serif.bold.default;
  } else if (italic) {
    fontStack = theme.typography.serif.regular.italic;
  }

  return {
    fontSize: theme.helpers.rem(1),
    lineHeight: theme.helpers.verticalRhythm(1, 1.625),
    fontFamily: fontStack,
    color: theme.colors.text.primary,
  };
}, 'ScriptureText');

const ScriptureText = compose(
  setDisplayName('ScriptureText'),
  styles,
  withPlaceholder(Typography),
  pure
)(Text);

ScriptureText.propTypes = {
  bold: PropTypes.bool,
  italic: PropTypes.bool,
  isLoading: PropTypes.bool, // display loading placeholder
  ...Text.propTypes,
};

ScriptureText.defaultProps = {
  bold: false,
  italic: false,
};

export default ScriptureText;
