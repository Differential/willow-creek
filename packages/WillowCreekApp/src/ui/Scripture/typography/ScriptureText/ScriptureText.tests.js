import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'WillowCreekApp/src/Providers';

import ScriptureText from '.';

describe('the ScriptureText component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Providers>
        <ScriptureText>Default ScriptureText text</ScriptureText>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render as bold', () => {
    const tree = renderer.create(
      <Providers>
        <ScriptureText bold>Bold ScriptureText text</ScriptureText>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render as italic', () => {
    const tree = renderer.create(
      <Providers>
        <ScriptureText italic>Italic ScriptureText text</ScriptureText>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render as bold italic', () => {
    const tree = renderer.create(
      <Providers>
        <ScriptureText bold italic>
          Bold italic ScriptureText text
        </ScriptureText>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmon = { color: 'salmon' };
    const tree = renderer.create(
      <Providers>
        <ScriptureText style={salmon}>Salmon text</ScriptureText>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state', () => {
    const tree = renderer.create(
      <Providers>
        <ScriptureText isLoading>Default ScriptureText text</ScriptureText>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept additional props', () => {
    const tree = renderer.create(
      <Providers>
        <ScriptureText accessible={false}>
          {
            '"True faith means holding nothing back. It means putting every hope in God\'s fidelity to His Promises." â€• Francis Chan'
          }
        </ScriptureText>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
