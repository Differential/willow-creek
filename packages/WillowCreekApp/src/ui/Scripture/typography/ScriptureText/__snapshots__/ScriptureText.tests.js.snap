// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ScriptureText component should accept additional props 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      accessible={false}
      bold={false}
      italic={false}
      style={
        Object {
          "color": "#303030",
          "fontFamily": "DroidSerif",
          "fontSize": 16,
          "lineHeight": 26,
        }
      }
    >
      "True faith means holding nothing back. It means putting every hope in God's fidelity to His Promises." â€• Francis Chan
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`the ScriptureText component should accept and render passed in styles 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      bold={false}
      italic={false}
      style={
        Object {
          "color": "salmon",
          "fontFamily": "DroidSerif",
          "fontSize": 16,
          "lineHeight": 26,
        }
      }
    >
      Salmon text
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`the ScriptureText component should render a loading state 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#A5A5A5",
          "borderRadius": 16,
          "height": 16,
          "marginVertical": 5,
          "width": "100%",
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;

exports[`the ScriptureText component should render as bold 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      bold={true}
      italic={false}
      style={
        Object {
          "color": "#303030",
          "fontFamily": "DroidSerif-Bold",
          "fontSize": 16,
          "lineHeight": 26,
        }
      }
    >
      Bold ScriptureText text
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`the ScriptureText component should render as bold italic 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      bold={true}
      italic={true}
      style={
        Object {
          "color": "#303030",
          "fontFamily": "DroidSerif-BoldItalic",
          "fontSize": 16,
          "lineHeight": 26,
        }
      }
    >
      Bold italic ScriptureText text
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`the ScriptureText component should render as italic 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      bold={false}
      italic={true}
      style={
        Object {
          "color": "#303030",
          "fontFamily": "DroidSerif-Italic",
          "fontSize": 16,
          "lineHeight": 26,
        }
      }
    >
      Italic ScriptureText text
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`the ScriptureText component should render correctly 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      bold={false}
      italic={false}
      style={
        Object {
          "color": "#303030",
          "fontFamily": "DroidSerif",
          "fontSize": 16,
          "lineHeight": 26,
        }
      }
    >
      Default ScriptureText text
    </Text>
  </View>
</RCTSafeAreaView>
`;
